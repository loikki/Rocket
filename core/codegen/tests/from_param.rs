use rocket::request::FromParam;

#[derive(Debug, FromParam, PartialEq)]
enum Test {
    Test1,
    Test2
}

#[test]
fn derive_from_param() {
    let test1 = Test::from_param("Test1").expect("Should be valid");
    assert_eq!(test1, Test::Test1);

    let test2 = Test::from_param("Test2").expect("Should be valid");
    assert_eq!(test2, Test::Test2);

    let test3 = Test::from_param("not_test");
    assert!(test3.is_err());
}
