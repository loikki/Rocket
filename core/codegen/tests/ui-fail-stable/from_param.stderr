error: named structs are not supported
 --> tests/ui-fail-stable/from_param.rs:4:1
  |
4 | struct Foo1 {
  | ^^^^^^

error: [note] error occurred while deriving `FromParam`
 --> tests/ui-fail-stable/from_param.rs:3:10
  |
3 | #[derive(FromParam)]
  |          ^^^^^^^^^
  |
  = note: this error originates in the derive macro `FromParam` (in Nightly builds, run with -Z macro-backtrace for more info)

error: named structs are not supported
 --> tests/ui-fail-stable/from_param.rs:9:1
  |
9 | struct Foo2 {}
  | ^^^^^^

error: [note] error occurred while deriving `FromParam`
 --> tests/ui-fail-stable/from_param.rs:8:10
  |
8 | #[derive(FromParam)]
  |          ^^^^^^^^^
  |
  = note: this error originates in the derive macro `FromParam` (in Nightly builds, run with -Z macro-backtrace for more info)

error: Only empty enums are accepted
  --> tests/ui-fail-stable/from_param.rs:13:6
   |
13 |     A(String),
   |      ^^^^^^^^

error: [note] error occurred while deriving `FromParam`
  --> tests/ui-fail-stable/from_param.rs:11:10
   |
11 | #[derive(FromParam)]
   |          ^^^^^^^^^
   |
   = note: this error originates in the derive macro `FromParam` (in Nightly builds, run with -Z macro-backtrace for more info)

error: tuple structs are not supported
  --> tests/ui-fail-stable/from_param.rs:21:1
   |
21 | struct Foo5(usize);
   | ^^^^^^

error: [note] error occurred while deriving `FromParam`
  --> tests/ui-fail-stable/from_param.rs:20:10
   |
20 | #[derive(FromParam)]
   |          ^^^^^^^^^
   |
   = note: this error originates in the derive macro `FromParam` (in Nightly builds, run with -Z macro-backtrace for more info)
